### ISSUE TITLE
remove the use of the --package-warnings flag in `flutter analyze`

### ISSUE THREAD

**Issue by devoncarew** [flutter/flutter#1680]
I don't think we want this flag - this causes all direct and transitively imported packages to be analyzed. We should remove it (and verify that all our packages in the flutter repo are still properly analzyed).

Related to https://github.com/dart-lang/sdk/issues/25722. /cc @Hixie. Happy to experiment w/ this myself (after closing out a few other PRs).

**Comment by Hixie** [comment]
Yeah I had turned this on because I wanted all _our_ direct and transitively imported packages to be analyzed. What I really want to be able to express is "analyze anything that didn't get pulled down by pub". For example, say we analyze a test in the stocks package. I want it to analyze all the stocks files, and all the flutter files that that pulls in, and all the files in the newton package that that pulls in. I'm worried that if we just rely on a test in the newton package pulling in the newton package, it'll actually only be bringing in the few files we happened to have tests for.

This gets less and less problematic as we add more and more tests, of course.

cc @pq
